src/python/pants/goal/help.py

```diff
@@ -84,7 +84,7 @@ class ThingHelpBuiltinGoal(HelpBuiltinGoalBase):
     def create_help_request(self, options: Options) -> HelpRequest:
         return ThingHelp(
             advanced=False,
-            things=tuple(options.goals) + tuple(options.unknown_goals),
+            things=tuple(options.goals) + tuple(options.unknown_goals) + tuple(options.specs),
```

Comment:
This change is to allow `./pants help foo.bar.baz`, as the `foo.bar.baz` is picked up as a spec.

Reply:
Huh, why is that the case? Because of the embedded `.` ?

Reply:
This merits a comment, because we're picking up something that isn't a spec into `specs`, and that is not obvious.

Reply:
Yeah, it's from arg splitter is_likely_a_spec looks for a range of special characters and file exists. But the most surprising and possibly error prone with that is that if you have a file named the same as a goal, it breaks the goal.